### PR TITLE
initial caliper annotations api

### PR DESCRIPTION
**Pull request by cyrush**
resolves #1045

### CODE REVIEW

**Comment by cyrush** [comment]
@agcapps initial pass at functional caliper setup w/ conduit annotation macros. 

we can expand as needed to wrap more caliper functionality, but I think the basics (region marking, function can scope marking) will take us a long way. 

For our tests we can use `conduit::annotation::initialize()` to capture info, host codes will likley have their own caliper setup.

Let me know what you think. I hope to merge soon and we can start using asap.

File: src/libs/conduit/conduit_utils.hpp [hunk 1]
```diff
@@ -21,12 +21,16 @@
...
 //-----------------------------------------------------------------------------
 // -- conduit includes --
 //-----------------------------------------------------------------------------
 #include "conduit_core.hpp"
 
+//-----------------------------------------------------------------------------
+#if defined(CONDUIT_USE_CALIPER)
+#include <caliper/cali.h>
+#endif
+
```
Comment (agcapps) [inline src/libs/conduit/conduit_utils.hpp]: Is conduit_utils.hpp a user-facing header file?  Will this let caliper leak out into a host code through Conduit?

Comment (cyrush) [reply src/libs/conduit/conduit_utils.hpp]: that one I need to remove, good catch!

**Comment by cyrush** [comment]
- [x] add caliper cmake exports and config.mk logic

**Comment by agcapps** [comment]
Thanks for this, Cyrus.  It looks good to me.  From my beginner's knowledge of Caliper, it does look like mark begin, end, function, and scope are reasonable things to expose.

**Review by agcapps — COMMENTED** [review]
Thanks, Cyrus.

**Pull request merged**